docs/02-developers/08-libraries/bitcoin-tx-solidity-helper.md

```diff
@@ -0,0 +1,168 @@
+| 8 | Value | `int64_t` | Number of satoshis to spend. May be zero; the sum of all outputs may not exceed the sum of satoshis previously spent to the outpoints provided in the input section. (Exception: coinbase transactions spend the block subsidy and collect transaction fees.) |
+| 1+ | pk_script bytes | compactSize uint | Number of bytes in the pubkey script. Maximum is 10,000 bytes. |
+| 1+ | pk_script | `char[]` | Defines the conditions which must be satisfied to spend this output. |
+
+> See the [Reference Implementation](https://developer.bitcoin.org/reference/transactions.html#txout-a-transaction-output)
+
+```solidity
+struct TxRawOutput {
+        uint64 value;
+        bytes pkScript;
+        uint256 scriptSize;
+        uint256 totalSize;
+    }
+```
+
+After finishing the processing of each output, the library returns an ordered output array, so the user can take advantage of this information in its solidity contract.  
+
+In order to show the benefits of this library, we’ll use the example of the [Flyover Protocol](/developers/integrate/flyover/). In this protocol, there is a smart contract that one party uses to claim a refund, in order to claim this refund, they need to prove that there was a payment with a specific amount done to a specific address in the Bitcoin Network, in order to do this, the smart contract receives the Bitcoin raw transaction. Since making this validation is not a trivial process, as it requires to parse the whole transaction, here is where we can see the utility of the library.
+
+The usage of the output parsing functionality is the following:
+```solidity
+BtcUtils.TxRawOutput[] memory outputs = BtcUtils.getOutputs(btcTx);
+```
+
+Then the user is able to perform any validation:
+```solidity
+require(expectedValue <= outputs[0].value, "incorrect amount");
+```
+
+:::info[Important]
+The value field of the output structure is in satoshis.
+:::
+
+## Hashing Transactions
+The hash algorithm used in the Bitcoin Network is just the `SHA256(SHA256())` of the serialized transaction. The library exposes one function that will apply this hash algorithm to any byte array passed to it, making it easy to calculate the transaction id of any raw transaction present in the contract.
+
+This function is specifically useful to interact with the [rootstock native bridge](/concepts/powpeg/), as many of its functions have a transaction id as parameter. For example, by using the transaction hash function, it is easy to know how many confirmations a Bitcoin block has inside a smart contract function.
+
+### Example code with explanation
+Based on the example stated in the previous section, after validating that a specific transaction has an output paying a certain amount to an address. We need to know if that transaction has enough confirmations:
+
+Here's an example:
+
+```solidity
+BtcUtils.TxRawOutput[] memory outputs = BtcUtils.getOutputs(btcTx);
+require(expectedValue <= outputs[0].value, "incorrect amount");
```

Comment:
Are you referring to this explanation @Luisfc68 ?

Reply:
no, to the "info" part, but I don't know if formatting this is supported by the PR preview so don't worry

<img width="1020" alt="image" src="https://github.com/user-attachments/assets/36cec099-cbe7-44cc-bbfa-f9754d2ac650">

Reply:
Ok. Cool